python/tvm/octo/octo_model.py

```diff
@@ -0,0 +1,191 @@
+    ) -> Tuple[relax.Executable, Dict[str, relax.StructInfo]]:
+        """Save the OctoModel to disk.
+
+        Parameters
+        ----------
+        model_path : Union[str, Path]
+            A full path to save this OctoModel to including the output file name.
+            The file will be saved as a tar file so using a ".tar" extension is advised.
+        """
+        # Only two artifacts need to be saved, the exe and the input struct info.
+        # Serialize both to a temp directory.
+        exe_path = self._tmp_dir.relpath("exe.so")
+        self.exe.mod.export_library(exe_path)
+        input_info_path = self._tmp_dir.relpath("input_info.json")
+        with open(input_info_path, "w") as fo:
+            json.dump(self.input_info, fo)
+
+        # Save additional metadata.
+        metadata = {"target": str(self.target)}
+        metadata_path = self._tmp_dir.relpath("metadata.json")
+        with open(metadata_path, "w") as fo:
+            json.dump(metadata, fo)
+
+        # Tar the tempfile and save to the designated model_path.
+        with tarfile.open(model_path, "w") as tar:
+            tar.add(exe_path, "exe.so")
+            tar.add(input_info_path, "input_info.json")
+            tar.add(metadata_path, "metadata.json")
+
+    def load(self, model_path: Union[str, Path]) -> Tuple[relax.Executable, Dict[List, str]]:
+        """Load a saved OctoModel back into memory.
+
+        Parameters
+        ----------
+        model_path : Union[str, Path]
+            The path to the saved OctoModel that will be loaded.
+
+        Returns
+        -------
+        exe : relax.Executable
+            A compiled executable that can be loaded and run by a relax VM.
+        input_info : Dict[str, Tuple[List, str]]
+            Information about the input names, shapes, and types for the VM.
+            Will be loaded from memory if possible.
+        """
+        t = tarfile.open(model_path)
+        t.extractall(self._tmp_dir.relpath("."))
+
+        # Load executable.
+        exe_path = self._tmp_dir.relpath("exe.so")
+        exe = relax.Executable(tvm.runtime.load_module(exe_path))
+
+        # Load input info.
+        input_info_path = self._tmp_dir.relpath("input_info.json")
+        with open(input_info_path, "r") as fi:
+            input_info = json.load(fi)
+
+        # load other metadata.
+        metadata_path = self._tmp_dir.relpath("metadata.json")
+        with open(metadata_path, "r") as fi:
+            metadata = json.load(fi)
+        self.target = tvm.target.Target(metadata["target"])
+
+        return exe, input_info
+
+    def generate_inputs(self) -> Dict[str, np.array]:
+        """Generate random inputs for inference or benchmarking
+
+        Returns
+        -------
+        input_dict : Dict[str, np.array]
+        """
+        input_dict = {}
+        for name, (shape, dtype) in self.input_info.items():
+            input_dict[name] = np.random.normal(size=shape).astype(dtype)
```

Comment:
Here the input could be integer, another alternative could be `tvm.meta_schedule.testing.tune_utils.generate_input_data`.

Reply:
oh neat, i hadnt seen that before.